### PR TITLE
feat: add toJSON method for good stringify

### PR DESCRIPTION
**Pull request by zxqfox**
cc @blond @kriant

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -222,6 +222,14 @@ module.exports = class BemEntityName {
         return `BemEntityName ${stringRepresentation}`;
     }
     /**
+     * Return raw data for JSON.stringify.
+     *
+     * @returns {*}
+     */
+    toJSON() {
```
Comment (skad0) [inline index.js]: When I call method `toJSON` I expect to have a json here. This method should be `getData` maybe.

Comment (zxqfox) [inline index.js]: It's for JSON.stringify

Comment (skad0) [inline index.js]: Why not to call JSON.stringify here if this is only for such purpose.

**Comment by blond** [comment]
Than this method differs from the `valueOf` method?

File: index.js [hunk 2]
```diff
@@ -222,6 +222,14 @@ module.exports = class BemEntityName {
         return `BemEntityName ${stringRepresentation}`;
     }
     /**
+     * Return raw data for JSON.stringify.
+     *
+     * @returns {*}
```
Comment (blond) [inline index.js]: Why `*`?

`this._data` has the schema:

```js
/**
 * @returns {{block: string, elem: ?string, mod: ?{name: ?string, val: *}}}
 */
```

File: index.js [hunk 3]
```diff
@@ -222,6 +222,14 @@ module.exports = class BemEntityName {
...
+     */
+    toJSON() {
+        return this._data;
```
Comment (blond) [inline index.js]: You want add alias? Then it is better to specify it explicitly:

```js
return this.valueOf();
```

**Comment by blond** [comment]
@zxqfox I think you need to use the `valueOf()` method. Reopen this PR if you have reason to have `toJSON` method.

**Pull request closed without merging**